auth_totp/controllers/main.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime, timedelta
+import json
+from werkzeug.contrib.securecookie import SecureCookie
+from openerp import http, registry, SUPERUSER_ID
+from openerp.api import Environment
+from openerp.http import Response, request
+from openerp.addons.web.controllers.main import Home
+from ..exceptions import MfaTokenInvalidError, MfaTokenExpiredError
+
+
+class JsonSecureCookie(SecureCookie):
+    serialization_method = json
+
+
+class AuthTotp(Home):
+    @http.route()
+    def web_login(self, *args, **kwargs):
+        response = super(AuthTotp, self).web_login(*args, **kwargs)
+
+        if not request.params.get('login_success'):
+            return response
+
+        user = request.env['res.users'].sudo().browse(request.uid)
+        if not user.mfa_enabled:
+            return response
+
+        cookie_key = 'trusted_devices_%s' % user.id
```

Comment:
Basically it's just the formatting language. In the example of English it's the difference between you saying "The Lake" and "Lake Tahoe". Sure both work, but one is more specific and the correct syntax grammatically.